internal/corazawaf/rule.go

```diff
@@ -512,6 +512,7 @@ func (r *Rule) executeTransformations(value string) (string, []error) {
 }
 
 // NewRule returns a new initialized rule
+// By default, the rule is set to phase 2
```

Comment:
I don't like much this implicit phase assignation, but I found some references [here](https://stackoverflow.com/questions/67316471/default-phase-in-modsecurity-rules) and it seems that we are sticking with Modsec 2 behavior (A rule without `phase` will default to `phase:2`).
Are we keeping it as it is for compatibility with ModSec? Are there strong reasons to not require an explicit phase?

Reply:
I also don't like this behavior. We should have explicit phases. And it should be documented properly.

Reply:
Would love a line from @airween

Reply:
Opened a dedicated issue to keep track of it https://github.com/corazawaf/coraza/issues/728